docs/developer/architecture/logs.rst

```diff
@@ -197,6 +197,447 @@ overview page.
+
+Collector
+~~~~~~~~~
+
+The collector is responsible for processing logs from all the sources
+(files, journal, containers, ...), enriching the logs with metadata (labels)
+coming from parsing/filtering them or by querying external sources such as
+APIs, then finally forwarding these logs to one or multiple distributors.
+
+Distributor (Router)
+~~~~~~~~~~~~~~~~~~~~
+
+The distributor is the component that receives incoming streams from the
+collectors, validates and prepares the streams for storage, then forwards
+them to the right ingester.
+The distributor can also do some processing on the streams to enrich them
+with metadata (labels).
+
+Ingester (Storage)
+~~~~~~~~~~~~~~~~~~
+
+The job of the ingester is to write log streams on persistent storage such
+as a filesystem or a database, but also to return the data requested by
+the queriers, if still in memory.
+
+Querier
+~~~~~~~
+
+The querier interprets the queries it receives from clients and then asks the
+ingesters for the corresponding data, then fallback on storage backend if not
+present in memory and returns it to the clients.
+It also takes care of deduplication of data because of the replication.
+
+Design Choices
+--------------
+
+To choose which solution fits the most our needs, we did a benchmark of some
+of the solutions to compare their performances.
+Our choice for the final design has been greatly motivated by these numbers,
+
+Check `this document`_ for details.
+
+.. _this document: https://docs.google.com/spreadsheets/d/1f2Pz9zBbkDzpHr8KzST3nTcfvWJqgoNcCEd0-4WIeow/edit#gid=0
+
+.. todo::
+
+  export the spreadsheet and include it or do directly a quick recap tab here.
+
+Fluent Bit + Fluentd (optional) + Loki
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Fluent Bit
+""""""""""
+
+We choose Fluent Bit as the collector because it allows to scrape all the logs
+we need (journal & containers), enriches them with the Kubernetes API and it
+has a very low resources footprint.
+
+Moreover, it supports multiple distributors such as Loki, ES, Splunk, etc. at
+the same time, which is a very important point if a user also wants to forward
+the logs to an external log centralization system (e.g. long term archiving).
+
+Check here_ for the official list of supported outputs.
+
+.. _here: https://docs.fluentbit.io/manual/pipeline/outputs
+
+Loki
+""""
+
+Loki has been chosen as the distributor, ingester & querier because, like
+Fluent Bit, it has a really small impact on resources and is very cost
+effective regarding storage needs.
+
+It also uses the LogQL syntax for queries, which is pretty close to what
+we already have with Prometheus and PromQL, so it eases the integration
+in our tools.
+
+Fluentd (optional)
+""""""""""""""""""
+
+Fluentd could be optionally added between Fluent Bit and Loki to offer
+more plugins and reduce the impact of Fluent Bit on workload nodes.
+
+It's not recommended for small platforms (few nodes) as it adds a big overhead
+in resources consumption.
```

Comment:
It's not clear what role Fluentd would play out in this case: Distributor, or a part of the Collector pipeline with Fluentbit?

For the "optional" part of this, is the design stating our implementation may include Fluentd, or is it saying we will have both options available in the product and expose this to our users?

Reply:
Oh yes, I agree, that's not crystal clear.
Fluentd will be part of the Collector pipeline, I'll explain it.

It means we will have both options available, depending on the needs, I will also try to explain a bit more.

Reply:
Fluentd has been removed from the design.